cmd/protoc-gen-openapi/generator/openapi-v3.go

```diff
@@ -623,10 +636,27 @@ func (g *OpenAPIv3Generator) buildOperationV3(
 		var requestSchema *v3.SchemaOrReference
 
 		if bodyField == "*" {
-			// Pass the entire request message as the request body.
-			typeName := fullMessageTypeName(inputMessage.Desc)
-			requestSchema = g.schemaOrReferenceForType(typeName)
+			if string(inputMessage.Desc.FullName()) != "google.api.HttpBody" && haveCommonItem(coveredParameters, g.getMessageFieldNames(inputMessage)) {
```

Comment:
do we have access to the actual type? may be more safe to check that against something that isn't hard coded